packages/edit-site/src/store/selectors.js

```diff
@@ -268,32 +324,11 @@ export const getCurrentTemplateTemplateParts = createRegistrySelector(
 			'wp_template_part',
 			{ per_page: -1 }
 		);
-		const templatePartsById = templateParts
-			? // Key template parts by their ID.
-			  templateParts.reduce(
-					( newTemplateParts, part ) => ( {
-						...newTemplateParts,
-						[ part.id ]: part,
-					} ),
-					{}
-			  )
-			: {};
-
-		return ( template.blocks ?? [] )
-			.filter( ( block ) => isTemplatePart( block ) )
-			.map( ( block ) => {
-				const {
-					attributes: { theme, slug },
-				} = block;
-				const templatePartId = `${ theme }//${ slug }`;
-				const templatePart = templatePartsById[ templatePartId ];
-
-				return {
-					templatePart,
-					block,
-				};
-			} )
-			.filter( ( { templatePart } ) => !! templatePart );
+
+		return memoizedGetFilteredTemplatePartBlocks(
+			template.blocks || [],
+			templateParts
+		);
```

Comment:
I'm guessing it's not possible to use createSelector with createRegistrySelector and that's why you had to rely on "memize" directly?

Reply:
Yes, that appears to be the case, unless I've missed something!